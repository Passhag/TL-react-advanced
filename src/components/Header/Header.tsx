import * as React from 'react';

interface Props {
  title: string;
}

const Header: React.SFC<Props> = ({ title }) => (
  <h1>{title}</h1>
);

export default Header;
