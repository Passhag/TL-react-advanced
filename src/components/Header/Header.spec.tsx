import * as React from 'react';
import { shallow, configure } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import Header from './Header';

configure({ adapter: new ReactSixteenAdapter() });

describe('Components::Header', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Header title={'List'} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
