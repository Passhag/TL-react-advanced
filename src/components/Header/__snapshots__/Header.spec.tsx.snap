// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::Header should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="List"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "List",
    },
    "ref": null,
    "rendered": "List",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "List",
      },
      "ref": null,
      "rendered": "List",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
